src/main/scala/edu/cmu/cs/obsidian/Main.scala

```diff
@@ -189,16 +192,7 @@ object Main {
 
             //copy the content of the fabric/java/ folder into a folder with the class name
             //have to add the trailing separator to avoid copying the java directory too
-            var compilerDir = System.getenv("TRAVIS_BUILD_DIR")
-            if (compilerDir == null) {
-                // TODO: package up the compiler as a jar file and use a path relative to that
-                // Requiring the working dir to be the compiler's directory is inconvenient.
-                compilerDir = System.getenv("OBSIDIAN_COMPILER_DIR")
-                if (compilerDir == null) {
-                    compilerDir = Paths.get("").toAbsolutePath().toString
-                }
-            }
-            val fabricPath = Paths.get(compilerDir, "fabric", "java")
+            val fabricPath = ImportProcessor.compilerPath().resolve("fabric").resolve("java")
```

Comment:
This seems like a weird place to put compilerPath. Maybe the method should be defined right here (in Main) and then passed as an argument when instantiating the ImportProcessor?